Import flow: Improve logic of showing loading screen

bogiii opened the pull request:
Related to #77719

## Proposed Changes

* Improved logic of showing loading screen fixing a moment of showing "NotFound" component

## Testing Instructions


https://github.com/Automattic/wp-calypso/assets/4074459/b89e22c6-8013-4841-84f7-4f06d943160c

See the demo video above. To reproduce this issue:
- Go to the site picker step and pick a site.
- Click the browser go back button, and pick another site.
- You'll see the Page not found component first, then wait a few seconds the page shows
## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

bogiii committed:
Import flow: Improve logic of showing loading screen

zaerl reviewed (approved):
LGTM.

ouikhuan reviewed (changes requested):
Looks like the behavior changed but I still see the page not found component. It looks like this:
- Showing loading screen
- Showing Page not found
- Showing loading screen
- Showing migrate ready page

https://github.com/Automattic/wp-calypso/assets/4074459/60f3f0aa-cdd2-446d-9ad2-fc5c969cc2e7

This is how I fix the same issue when clicking goBack button in the page, maybe we can consider reset this when the component is unmounted:
https://github.com/Automattic/wp-calypso/blob/bb07c7d43f1d4ff77be619207eeafd3c108b1f4c/client/landing/stepper/declarative-flow/internals/steps-repository/importer/index.tsx#L108

bogiii committed:
Importer: dispatch resetImportReceived action on component unmount

bogiii committed:
Importer: remove redundant action dispatch; the component unmount will handle it

bogiii commented:
I was testing between two simple sites, and in that case, it was working fine. Thanks for the review!

>This is how I fix the same issue when clicking goBack button in the page, maybe we can consider reset this when the component is unmounted:

@ouikhuan, thanks for proposing the solution. I've added the componentUnmount handler, and it seems it works correctly now.
Also, I've removed redundant dispatch from `onBack` handler since there is a dedicated `compnentUnmount` handler.

ouikhuan reviewed (approved):
LGTM!
